pkg/util/k8s/k8s.go

```diff
@@ -1992,6 +1992,31 @@ func CreateOrUpdateSecret(
 	return nil
 }
 
+// GetSecret get a secret from k8s
+func GetSecret(
+	k8sClient client.Client,
+	name string,
+	namsespace string,
+	secret *v1.Secret,
+) error {
+	err := k8sClient.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      name,
+			Namespace: namsespace,
+		},
+		secret,
+	)
+
+	if errors.IsNotFound(err) {
+		return nil
```

Comment:
Is it a common practice in Operator to swallow NotFound error? I would find this very misleading. If GetSecret() returns err==nil, I would expect that it has found the secret I asked it to get.